### PR TITLE
Update docs for RestoreBatch + Other improvements

### PR DESCRIPTION
**Pull request by hossainemruz**
Tasks:
- [x] Update Concepts docs
    - [x] README.md (add BackupBatch and RestoreBatch)
    - [x] overview.md (update feature table)
    - [x] architecture.md (fix heading, add RestoreBatch reference)
    - [x] Update CRD docs
      - [x] BackupConfiguration (alias, exclude field + status)
      - [x] BackupBatch(alias, exclude, executionOrder field + status)
      - [x] BackupSession ( status)
      - [x] RestoreSession (alias, rules, status)
      - [x] RestoreBatch
      - [x] Function (volumeMounts in `update-status` function)
      - [x] Task (secretVolume in `update-status` step)
      - [x] Snapshots (commands, status, pre-conditions)
- [x] Update NFS backend doc (running Stash operator in privileged mode + configuring NFS helper)
- [x] Update batch backup & restore doc
   - [x] How restore batch works?
   - [x] Batch backup and batch restore
   - [x] Batch backup and individual restore
   - [x] Execution order example
- [x] Update the YAMLs of all backup and restore tutorials. (alias + restore rules)

### CODE REVIEW

File: docs/concepts/crds/snapshot.md [hunk 1]
```diff
@@ -98,19 +102,25 @@ Here, we are going to describe the various sections of a `Snapshot` object.
 
 ```console
 # List Snapshots of all Repositories in the current namespace
-$ kubectl get snapshot
+$ kubectl get snapshot --request-timeout=300s
```
Comment (tamalsaha) [inline docs/concepts/crds/snapshot.md:105]: Explain why timeout is necessar.

File: docs/guides/latest/backends/local.md [hunk 2]
```diff
@@ -129,6 +131,8 @@ $ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version"
 repository/local-repo-with-nfs created
 ```
 
+>For NFS backend, Stash may have to run the network volume accessor deployments in privileged mode to provide Snapshot listing facility. In this case, please configure network volume accessors by following the instruction [here](/docs/setup/install.md#configuring-network-volume-accessor).
+
```
Comment (tamalsaha) [inline docs/guides/latest/backends/local.md:135]: `(/docs/setup/install.md#configuring-network-volume-accessor)` is not a valid link

**Comment by tamalsaha** [comment]
I think we should rewrite the warning line so that we can just copy paste it.

```
{{< notice type="warning" message="This is an Enterprise only feature. Please install [Stash Enterprise Edition](Link) to try this feature." >}}
```

File: docs/guides/latest/batch-backup/overview.md [hunk 3]
```diff
@@ -45,10 +47,26 @@ The backup process consists of the following steps:
...
+The following diagram shows the batch restore process. Please, open image in new tab to view the enlarged image.
+
+<figure align="center">
+  <img alt="Stash Batch Backup Flow" src="/docs/images/guides/latest/batch-backup/batch-restore.svg">
+<figcaption align="center">Fig: Batch restore flow in Stash</figcaption>
+</figure>
+
+The batch restore process consists of the following steps:
+
+1. At first, the user creates a `RestoreBatch` CR specifying the targets and the respective Repository where the backed up data has been stored.
```
Comment (tamalsaha) [inline docs/guides/latest/batch-backup/overview.md:65]: Should we mention the hooks?

File: docs/guides/latest/batch-backup/batch-backup.md [hunk 4]
```diff
@@ -489,170 +506,450 @@ spec:
...
 
-When a backup session is completed, Stash will update the respective `Repository` to reflect the latest state of backed up data.
+- `spec.repository` specifies that we are restoring from the `gcs-repo` Repository.
+- `spec.executionOrder` specify that we want Stash to restore the component sequentially. Here, we want to restore the database first then we want to restore the wordpress deployment's data.
+- `spec.members` specify the targets to be restored. Each member may have the following sub-fields.
+  - `target.alias` specify the host identifier of the backed up data for this member. It must be the same as the `alias` used during backup.
+  - `target.ref` refers to the target to restore.
+  - `target.rules` specify the rules for restoring the data of this member. Here, we want to restore the `latest` snapshot for the database and the latest state of `/var/www/html` path for our WordPress deployment.
+  - `target.volumeMounts` specifies the volume mounts where the data will be restored.
+  - `task.name` refers to the `Task` object to use to restore the member in the Function-Task model.
```
Comment (tamalsaha) [inline docs/guides/latest/batch-backup/batch-backup.md:779]: Should we also mention task.params ?

Comment (hossainemruz) [reply docs/guides/latest/batch-backup/batch-backup.md:779]: No. This example does not utilize that field. `task.params` has been explained in concept docs.

**Pull request merged**